node/src/actor_system_factory.rs

```diff
@@ -1977,17 +1999,78 @@ mod tests {
+                            .to_string(),
+                    ),
+                    block_number: Some("0x7D0".to_string()), // 2000 decimal
+                    address: "0x3333333333333333333333333333333333333334".to_string(),
+                    data: "0x000000000000000000000000000000000000000000000000000000003b5dc100"
+                        .to_string(),
+                    topics: vec![
+                        "0xddf252ad1be2c89b69c2b0680000000000006561726e696e675f77616c6c6574"
+                            .to_string(),
+                        "0xddf252ad1be2c89b69c2b0690000000000006561726e696e675f77616c6c6574"
+                            .to_string(),
+                    ],
+                }],
+                1,
+            )
+            .start();
+        let server_url = format!("http://{}:{}", &Ipv4Addr::LOCALHOST, port);
+        let _persistent_config = {
+            let conn = DbInitializerReal::default()
+                .initialize(&data_dir, DbInitializationConfig::test_default())
+                .unwrap();
+            PersistentConfigurationReal::from(conn)
         };
-
-        test_blockchain_bridge_is_constructed_with_correctly_functioning_connections(
-            "actor_system_factory",
-            "blockchain_bridge_is_constructed_with_correctly_functioning_connections",
-            act,
-        )
+        let wallet = make_wallet("abc");
+        let mut bootstrapper_config = BootstrapperConfig::new();
+        bootstrapper_config
+            .blockchain_bridge_config
+            .blockchain_service_url_opt = Some(server_url);
+        bootstrapper_config.blockchain_bridge_config.chain = TEST_DEFAULT_CHAIN;
+        bootstrapper_config.data_directory = data_dir.clone();
+        let system = System::new(test_name);
+        let (accountant, _, accountant_recording) = make_recorder();
+        let accountant = accountant.system_stop_conditions(match_every_type_id!(ReceivedPayments));
+        let peer_actors = peer_actors_builder().accountant(accountant).build();
+        let (tx, blockchain_bridge_addr_rx) = bounded(1);
+        let address_leaker = SubsFactoryTestAddrLeaker { address_leaker: tx };
+
+        ActorFactoryReal::new()
+            .make_and_start_blockchain_bridge(&bootstrapper_config, &address_leaker);
+
+        let blockchain_bridge_addr = blockchain_bridge_addr_rx.try_recv().unwrap();
+        blockchain_bridge_addr
+            .try_send(BindMessage {
+                peer_actors: peer_actors,
+            })
+            .unwrap();
+        blockchain_bridge_addr
+            .try_send(RetrieveTransactions {
+                recipient: wallet,
+                response_skeleton_opt: None,
+            })
+            .unwrap();
+        assert_eq!(system.run(), 0);
+        let recording = accountant_recording.lock().unwrap();
+        let received_payments_message = recording.get_record::<ReceivedPayments>(0);
+        assert!(received_payments_message.scan_result.is_ok());
```

Comment:
I was surprised when I saw you guys decided to rework this test. It's one of the most spiteful tests in the codebase, because it wasn't possible to write it elegantly. When I tried for that, I was so delved in the code that I might've forgotten to consider if the goal, to be sure about the two connections (BlockchainInterface - http, PersistentConfig - DB) being functional, hadn't been already included in some other, much higher level tests, and so I'm speaking of the integration ones actually. It would be the more elegant solution I touched, because it would feel smoothing if we could just put this unit test into the garbage because it even seems not to fit in well among the others (except the one just above, which is the same kind of bastard - ironically, also written in my person). So, that's the question... In an ideal word, I think, there should be at least one integration test which would do this favor for us. I'm not so positive there is. 

Also, this test is an attempt to have an extremely responsible coverage, maybe beyond how much we should be willing to do for this. 
The test was supposed to assure us that if we construct and start the BlockchainBridge, it will end up connected properly two two points. The correct database (file), and the correct Internet domain. 

If you do want to keep this test (in other words, if you don't want to discard in on your responsibility), you'll need to do something about the assertions. Now they are just incomplete. 

For the blockchain service endpoint, the assertion would be strongest if you could pull the received request out of the mock server and use it to prove the Actor sent it out. 
For the database, which must basically be about only the config table (because the blockchain bridge has only such an access, via the PersistentConfiguration), you'd have to prove that the database changed the state, which is hard actually. 

Because you moved the test body here from BlockchainBridge, you lost the advantage of being able to use the AssertionMessage and check whatever you want from inside the Actor. It's a nice trick, but the technical side of keeping the fields of the Actor private and still testing such things somewhere outside the Actor's home file, it's all pain. 

To be honest, under your new setting I can't think of something what I could advise you. (I know you could theoretically respond by an error to the blockchain scan, saying the requested block range is too wide, because as far as I can recall, that results in using the PersistentConfig's job of writing the parsed max range into the database, and you could check the presence of the value in there, when the test is over). 
 
You can have your private brainstorming over this or you can bring it up in the all-heads meeting. :man_shrugging:

Reply:
Would you accept adding further assertions to the `received_payments_message` to increase the strength of this test? or would you just prefer we delete this test as it could be misleading?

Reply:
Hi buddy, I pondered how I could help you out and despite I'm in general into testing these niches that are left behind and ignored. What I'm ragging you with is what simply comes across as an integration test that should probably really prove that the blockchain bridge is connected to the right outside points that it depends on somehow. If further looked into, the both outside technologies meet in the receivable scanner cycle. 

I first thought I'd give you advice which would be good enough to test it here. I just couldn't shake off the feeling that it would've been copying tightly unit tests testing those collaboration procedures with other processes, and the test itself would end up complicated, atypical and therefore rather confusing, and last but not least also a bit awkward. 

So I then checked my other thought of integration tests we already have and it appeased me enough to say that we can stay without this test and you can delete it. The test (or tests in a plural since GH-711) in  verify_bill_payments is probably the only one which actually works as an evidence of these connections being properly maintained and functioning. As long as that multimode test exists it's not so terrible important to have another test for a more hardened test coverage. 

I'd be glad if you understood my standpoint at least partially, and took something away, as good programming is paved by building hypothesis that are, as time passes, more often demolished then found really useful. Hurrah. 

(In short: Delete the whole test)